js/repl/src/repl/Repl.js

```diff
@@ -0,0 +1,327 @@
+// @flow
+
+import { css } from 'glamor';
+import React from 'react';
+import CodeMirrorPanel from './CodeMirrorPanel';
+import ReplOptions from './ReplOptions';
+import StorageService from './StorageService';
+import UriUtils from './UriUtils';
+import compile from './compile';
+import loadPlugin from './loadPlugin';
+import { pluginConfigs, presetPluginConfigs } from './PluginConfig';
+import { media } from './styles';
+
+import type { PersistedState, PluginConfigs, PluginStateMap } from './types';
+
+type Props = {};
+type State = {
+  code: string,
+  compiled: ?string,
+  compileError: ?Error,
+  evalError: ?Error,
+  evaluate: boolean,
+  lineWrap: boolean,
+  plugins: PluginStateMap,
+  presets: PluginStateMap
+};
+
+export default class Repl extends React.Component {
+  static defaultProps = {
+    defaultValue: ''
+  };
+
+  props: Props;
+  state: State;
+
+  _numLoadingPlugins = 0;
+
+  constructor(props: Props, context: any) {
+    super(props, context);
+
+    const persistedState = this._loadState();
+    console.log(
+      'persistedState.lineWrap:',
+      persistedState.lineWrap,
+      typeof persistedState.lineWrap,
+      persistedState
+    );
+
+    const defaultPlugins = {
+      'babili-standalone': persistedState.babili,
+      prettier: persistedState.prettier
+    };
+
+    const defaultPresets = (persistedState.presets || 'es2015,react,stage-2')
+      .split(',')
+      .reduce((reduced, key) => {
+        reduced[`babel-preset-${key}`] = true;
+        return reduced;
+      }, {});
+
+    // TODO Parse babel-preset-env settings
+    const state = {
+      code: persistedState.code,
+      compiled: null,
+      compileError: null,
+      evalError: null,
+      evaluate: persistedState.evaluate,
+      lineWrap: persistedState.lineWrap,
+      plugins: configToState(pluginConfigs, false, defaultPlugins),
+      presets: configToState(presetPluginConfigs, true, defaultPresets)
+    };
+
+    this.state = {
+      ...state,
+      ...this._compile(persistedState.code, state)
+    };
+    console.log(
+      'state.lineWrap:',
+      this.state.lineWrap,
+      typeof this.state.lineWrap,
+      persistedState
+    );
+
+    // Load any plug-ins enabled by query params
+    this._checkForUnloadedPlugins();
+  }
+
+  render() {
+    const {
+      code,
+      compiled,
+      compileError,
+      evaluate,
+      evalError,
+      lineWrap,
+      plugins,
+      presets
+    } = this.state;
```

Comment:
This is just a habit, partly for consistency and partly for aesthetics, but I agree it's unnecessary. It's also the sort of thing I hate to have to think about while coding. 😄